single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultLabelProvider.java

```diff
@@ -215,6 +194,39 @@ private List<FeederInfo> buildFeederInfos(Terminal terminal) {
         return buildFeederInfos(terminal, true);
     }
 
+    private List<FeederInfo> getBranchFeederInfos(Branch<?> branch, TwoSides side) {
+        List<FeederInfo> feederInfoList = buildFeederInfos(branch.getTerminal(side), true);
+        if (this.svgParameters.isDisplayPermanentLimitPercentageFeederInfo()) {
+            feederInfoList.add(new ValueFeederInfo(VALUE_PERMANENT_LIMIT_PERCENTAGE, getPermanentLimitPercentageMax(branch), valueFormatter::formatPercentage));
+        }
+        return feederInfoList;
+    }
+
+    private List<FeederInfo> get3WTFeederInfos(ThreeWindingsTransformer transformer, ThreeSides side, boolean insideVoltageLevel) {
+        List<FeederInfo> feederInfoList = buildFeederInfos(transformer.getTerminal(side), insideVoltageLevel);
+        if (this.svgParameters.isDisplayPermanentLimitPercentageFeederInfo()) {
+            feederInfoList.add(new ValueFeederInfo(VALUE_PERMANENT_LIMIT_PERCENTAGE, getPermanentLimitPercentageMax(transformer), valueFormatter::formatPercentage));
+        }
+        return feederInfoList;
+    }
+
+    private double getPermanentLimitPercentageMax(Branch<?> branch) {
+        return Arrays.stream(TwoSides.values())
+            .map(side -> getPermanentLimitPercentageMax(branch.getTerminal(side), branch.getCurrentLimits(side)))
+            .mapToDouble(Double::doubleValue).max().getAsDouble();
+    }
+
+    private double getPermanentLimitPercentageMax(ThreeWindingsTransformer transformer) {
+        return Arrays.stream(ThreeSides.values())
+            .map(side -> getPermanentLimitPercentageMax(transformer.getTerminal(side), transformer.getLeg(side).getCurrentLimits()))
+            .mapToDouble(Double::doubleValue).max().getAsDouble();
+    }
+
+    private double getPermanentLimitPercentageMax(Terminal terminal, Optional<CurrentLimits> currentLimits) {
```

Comment:
You should not use Optional as method or class parameter. Optional is primarily intended for use as a method return type where there is a clear need to represent "no result".